src/server.py

```diff
@@ -21,23 +37,61 @@ async def api_v1():
 async def api_v1_deal():
     return {"rank": random.choice(RANKS), "suit": random.choice(SUITS)}
 
-
+#creates new deck, adds deck ID and associated deck object to dictionary, generates new expiration time associated with it
 @app.post("/api/v2/deck/new")
 async def api_v2_deck_new():
     d = Deck()
-    return {"message": "deck created!"}
-
+    deck_id = generate_unique_deck_id()
+    deck_objects[deck_id] = d
+    expiration_time = time.mktime((datetime.now() + timedelta(hours=24)).timetuple()) #converts datetime to unix timestamp
+    expiration_times[deck_id] = expiration_time
+    hand = d.draw(5)  
+    return {
+            "deck_id": deck_id,
+            "expires": expiration_time,
+            "hand": { 
+                "cards": [hand]
+                }
+            }
 
+#returns deck associated with given parameter deck_id
 @app.get("/api/v2/deck/{deck_id}")
 async def api_v2_deck(deck_id: str):
-    print(f"need to fetch Deck {deck_id}")
-    raise HTTPException(status_code=404, detail=f"Deck {deck_id} not found")
+    if(deck_id not in deck_objects):
+        raise HTTPException(status_code=404, detail=f"Deck {deck_id} not found")
+    return deck_objects.get(deck_id)
+
+#returns cards equal to given count from the top of the deck
+#TODO: check for valid deals
+@app.post("/api/v2/deck/{deck_id}/deal/{count}")
+async def api_v2_deck_deal(deck_id: str, count: int):
+    if(deck_id not in deck_objects):
+        raise HTTPException(status_code=404, detail=f"Deck {deck_id} not found")
+    d = deck_objects.get(deck_id)
+    dealt_cards = d.draw(count)
+    return {
+        "dealt_cards": { 
+        "cards": [dealt_cards] 
+        }
+    }
```

Comment:
I am confused as to why we have `cards` wrapped in `dealt_cards`. This is workable, though, I suppose.

Reply:
Yeah, this is probably more complex than it's worth. You could do something like this:

```
return {
  "dealt_cards": dealt_cards,
  "count": count
}
```

That might be easier to work with on the client.